File: Leetcode/1266. Minimum Time Visiting All Points/solution.go
func minTimeToVisitAllPoints(points [][]int) int {
	if len(points) == 1 {
		return 0
	}

	total := 0

	for i := 1; i < len(points); i += 1 {
		last := points[i-1]
		current := points[i]
		difX := abs(current[0] - last[0])
		difY := abs(current[1] - last[1])
		total += max(difX, difY)
	}
	return total
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

File: Leetcode/1266. Minimum Time Visiting All Points/solution.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        if len(points) == 1:
            return 0
        time = 0
        for i in range(1, len(points)):
            current = points[i]
            last = points[i - 1]
            dif_x = abs(current[0] - last[0])
            dif_y = abs(current[1] - last[1])
            time += max(dif_x, dif_y)
        return time
